solana/modules/wormhole-anchor-sdk/src/token_bridge/message.rs

```diff
@@ -22,35 +22,51 @@ pub struct TransferWithMeta {
 }
 
 impl AnchorDeserialize for TransferWithMeta {
-    fn deserialize(buf: &mut &[u8]) -> io::Result<Self> {
+    fn deserialize_reader<R: io::Read>(reader: &mut R) -> io::Result<Self> {
+        let mut variant = 0;
+        reader.read_exact(slice::from_mut(&mut variant))?;
         // Verify Payload ID is a token transfer with payload.
-        if buf[0] != PAYLOAD_ID_TRANSFER_WITH_PAYLOAD {
+        if variant != PAYLOAD_ID_TRANSFER_WITH_PAYLOAD {
             return Err(io::Error::new(
                 io::ErrorKind::InvalidInput,
                 "Invalid Token Bridge Transfer With Payload",
             ));
         }
 
+        // Skip the next 24 bytes:
+        {
+            let mut _discard = [0u8; 32];
```

Comment:
The comment and the number are at odds. What are the extra 8 bytes?

Reply:
I'm using read now, as suggested below.